edb/server/tenant.py

```diff
@@ -358,13 +372,15 @@ async def init(self) -> None:
 
         if self._readiness_state_file is not None:
 
-            def reload_state_file(_file_modified, _event):
-                self.reload_readiness_state()
-
             self.reload_readiness_state()
-            self._server.monitor_fs(
-                self._readiness_state_file, reload_state_file
-            )
+
+            if not os.getenv("EDGEDB_SERVER_DISABLE_MONITOR_FS"):
```

Comment:
This should be something like `EDGEDB_SERVER_CONFIG_FILE_RELOAD` and be an enum (`none`, `signal`, `fsevent`).

Reply:
Yeah I heard you said that in our meeting too - TL; DR: I'm just concerned this may be a breaking change and decided to add a less intrusive "internal" env var.

Currently, we have the following types of "config files":

1. TLS certificate and key (server)
2. JWS key (server)
3. Readiness state (tenant)
4. JWT sub allowlist and revocation list (tenant)

And before this PR, the reloading condition is like:

| Config file  | Single-tenant | Multi-tenant |
|--------------|--------------------|-------------------|
| 1. TLS       | signal, fsevent    | signal, fsevent   |
| 2. JWS key   | signal             | signal            |
| 3. Readiness | signal, fsevent    | fsevent           |
| 4. JWT lists | signal             | none              |

(This PR adds the missing `signal` to 3 and 4 under multi-tenant mode and allow stripping `fsevent`.)

My concern is, if `--config-file-reload` is a single global enum (defaults to `signal`), we'll change the default dual reload of 1. TLS and 3. Readiness for regular instances. It's probably not a problem as we didn't advertise the `fsevent` reload as a stable feature, but it's been out for a while. So if you think that's okay, I can add this enum as a proper option/env var and add `fsevent` to 2. JWS key and 4. JWT lists.

Reply:
The default value for the env var should be `"default"` and it should mimic the pre-PR behavior.